src/custom/utils/paraswap.ts

```diff
@@ -0,0 +1,91 @@
+import { OrderKind } from '@gnosis.pm/gp-v2-contracts'
+import { ParaSwap, SwapSide, NetworkID } from 'paraswap'
+import { toErc20Address } from 'utils/tokens'
+import { PriceQuoteParams } from 'utils/operator'
+import { OptimalRatesWithPartnerFees, APIError, RateOptions } from 'paraswap/build/types'
+import { ChainId } from '@uniswap/sdk'
+import { PriceInformation } from 'state/price/reducer'
+import { getTokensFromMarket } from './misc'
+
+type ParaSwapPriceQuote = OptimalRatesWithPartnerFees
+
+// Provided manually just to make sure it matches what GPv2 backend is using, although the value used  is the current SDK default
+const API_URL = 'https://apiv4.paraswap.io/v2'
+
+const paraSwapLibs: Map<ChainId, ParaSwap> = new Map()
+
+function getParaswapChainId(chainId: ChainId): NetworkID | null {
+  switch (chainId) {
+    // Only Mainnnet and Ropsten supported
+    //  See https://developers.paraswap.network/api/list-all-tokens
+    case ChainId.MAINNET:
+    case ChainId.ROPSTEN:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
+}
+
+export function toPriceInformation(priceRaw: ParaSwapPriceQuote | null): PriceInformation | null {
+  if (!priceRaw || !priceRaw.details) {
+    return null
+  }
+
+  const { destAmount, srcAmount, details, side } = priceRaw
+  if (side === SwapSide.SELL) {
+    return {
+      amount: destAmount,
+      token: details.tokenTo
+    }
+  } else {
+    return {
+      amount: srcAmount,
+      token: details.tokenFrom
+    }
+  }
+}
+
+function isGetRateSuccess(
+  rateResult: OptimalRatesWithPartnerFees | APIError
+): rateResult is OptimalRatesWithPartnerFees {
+  return !!(rateResult as OptimalRatesWithPartnerFees).destAmount
+}
+
+export async function getPriceQuote(params: PriceQuoteParams): Promise<ParaSwapPriceQuote | null> {
+  const { baseToken: baseTokenAux, quoteToken: quoteTokenAux, amount, kind, chainId } = params
+  const baseToken = toErc20Address(baseTokenAux, chainId)
+  const quoteToken = toErc20Address(quoteTokenAux, chainId)
+
+  let paraSwap = paraSwapLibs.get(chainId)
```

Comment:
nit: i would move this logic above the `const` declaration for base/quoteToken in case it returns null. you don't use the constants until later anyways